### PR TITLE
#localdirectorystore get pathlike

### PR DESCRIPTION
**Pull request by nimshi89**
### What

The `get_pathlike_of_file_matching` function has been created in `local.py`
and a test for it has been made in `test_local_directory.py`

### How to review

run the test and check if it functions how it was intended

### Who can review

anyone

### CODE REVIEW

File: dpytools/stores/directory/local.py [hunk 1]
```diff
@@ -74,6 +74,23 @@ def has_lone_file_matching(self, pattern: str) -> bool:
                 f"More than 1 file found that matches the regex pattern '{pattern}' in directory {self.local_path}. Matching: {matching_files}"
             )
 
+    def get_pathlike_of_file_matching(self, pattern: str):
```
Comment (SarahJohnsonONS) [inline dpytools/stores/directory/local.py]: Add `get_pathlike_of_file_matching` abstract method to the `BaseReadableSingleDirectoryStore` class in `stores/directory/base.py`

File: dpytools/stores/directory/local.py [hunk 2]
```diff
@@ -74,6 +74,23 @@ def has_lone_file_matching(self, pattern: str) -> bool:
...
+        """
```
Comment (SarahJohnsonONS) [inline dpytools/stores/directory/local.py:78]: Add a return type of Path to `get_pathlike_of_file_matching`

File: dpytools/stores/directory/local.py [hunk 3]
```diff
@@ -74,6 +74,23 @@ def has_lone_file_matching(self, pattern: str) -> bool:
...
+            raise FileNotFoundError(
+                f"No matching files found for pattern {pattern} in directory {self.local_path}"
+            )
+        
+        matching_file_name = self._files_that_match_pattern(pattern)[0]
+
+        # Assemble the full path
+        matching_file_full_path = os.path.join(self.local_path, matching_file_name)
+
+        return matching_file_full_path
```
Comment (SarahJohnsonONS) [inline dpytools/stores/directory/local.py]: Convert `matching_file_full_path` to a `pathlib.Path` instead of a string, and update the test as required.

**Review by SarahJohnsonONS — COMMENTED** [review]
Comments added

File: dpytools/stores/directory/base.py [hunk 4]
```diff
@@ -58,6 +58,13 @@ def get_current_source_pathlike(self) -> str:
         Get the source in terms of the current path like location as a string representing the directory/bucket-like the files are in.
         """
         ...
+    
+    @abstractmethod
+    def get_pathlike_of_file_matching(self, pattern: str) -> str:
```
Comment (SarahJohnsonONS) [inline dpytools/stores/directory/base.py]: Change return type to Path

Comment (nimshi89) [reply dpytools/stores/directory/base.py]: my bad sorry :D

**Review by SarahJohnsonONS — APPROVED** [review]
LGTM

**Pull request merged**